### ISSUE TITLE
A issue about torch.optim.lr_scheduler

### ISSUE THREAD

**Issue by ZyUestc** [pytorch/pytorch#33416]
## 🐛 Bug

I want to use 2 scheduler during training. First to warmup learning rate, initial rate is 1e-4, and double it per epoch. Second to decay rate, half it per epoch. I found the initial rate of second would not be set to learning rate of optimizer correctly and where the problem is. 

## To Reproduce

```
def dataloader_worker(pId, gpuNum, dataset, batchSize):
    dist.init_process_group(backend='gloo', init_method='tcp://127.0.0.1:6666', world_size=gpuNum ,rank=pId)
    
    optimizer = optim.Adam([torch.rand(10, 20),], lr=1e-4)
    
    warmup_factor = 2
    decay_factor = 0.5
    
    warmup_schedular = optim.lr_scheduler.LambdaLR(optimizer, lambda epoch : warmup_factor ** epoch) 
    
    sampler = data.distributed.DistributedSampler(dataset)
    dataloader = data.DataLoader(dataset, batch_size=batchSize, sampler=sampler)
    
    schedular = warmup_schedular
    for epoch in range(4):
        if epoch == 2:
            # print('lr : ', optimizer.param_groups[0]['lr'])
            # print('inital lr : ', optimizer.param_groups[0]['initial_lr'])
            schedular = optim.lr_scheduler.LambdaLR(optimizer, lambda epoch : decay_factor ** epoch)
            # print('base lr  : ', schedular.base_lrs)
            # print('inital lr : ', optimizer.param_groups[0]['initial_lr'])
        for x in dataloader:
            optimizer.step()
        schedular.step()
        print(pId, ':', optimizer.param_groups[0]['lr'])
```

```
def test():
    gpuNum=1
    batchSize=16
    X = torch.arange(16)
    dataset = data.TensorDataset(X)
    mp.spawn(dataloader_worker, nprocs=gpuNum, args=(gpuNum, dataset, batchSize))
```

## Expected behavior

0 : 0.0002
0 : 0.0004
0 : 0.0002
0 : 0.0001

## Environment

PyTorch 1.3.1

You can get the script and run it with:
`test()`

## Additional context

I found problem in [source code ](https://pytorch.org/docs/stable/_modules/torch/optim/lr_scheduler.html#LambdaLR).
```
class _LRScheduler(object):

   def __init__(self, optimizer, last_epoch=-1):
        # blabla
        # Initialize epoch and base learning rates
        if last_epoch == -1:
            for group in optimizer.param_groups:
                group.setdefault('initial_lr', group['lr'])
```

if you use a sceduler before, group in optimizer.param_groups has a 'initial' already, so the 'initial_lr'  will not be set correctly. 
I note off code in 'if epoch == 2', and get:
lr :  0.0004
inital lr :  0.0001
base lr  :  [0.0001]
inital lr :  0.0001

I understand this code is for resuming,  but that will produce unexpected results if use 2 scheduler. So I wonder there should have some code for estimating if the scheduler is for resuming, or allow user to set initial learning rate.

cc @vincentqb

**Comment by vincentqb** [comment]
Simpler example without multiprocessing:
```python
import torch
from torch import optim
from torch.utils import data

X = torch.arange(16) 
dataset = data.TensorDataset(X) 

optimizer = optim.Adam([torch.rand(10, 20),], lr=1e-4) 

warmup_factor = 2 
decay_factor = 0.5 

warmup_schedular = optim.lr_scheduler.LambdaLR(optimizer, lambda epoch : warmup_factor ** epoch)  

batchSize = 16
dataloader = data.DataLoader(dataset, batch_size=batchSize) 

schedular = warmup_schedular 

for epoch in range(4): 
  if epoch == 2: 
    schedular = optim.lr_scheduler.LambdaLR(optimizer, lambda epoch : decay_factor ** epoch) 
  for x in dataloader: 
      optimizer.step() 
  schedular.step() 
  print(optimizer.param_groups[0]['lr']) 
```
```
0.0002
0.0004
5e-05
2.5e-05
```

**Comment by vincentqb** [comment]
You can user `MultiplicativeLR` for this.

```python
import torch
from torch import optim
from torch.utils import data

X = torch.arange(16) 
dataset = data.TensorDataset(X) 

optimizer = optim.Adam([torch.rand(10, 20),], lr=2e-4) 

warmup_factor = 2 
decay_factor = 0.5 

scheduler1 = optim.lr_scheduler.MultiplicativeLR(optimizer, lambda epoch : warmup_factor)  
scheduler2 = optim.lr_scheduler.MultiplicativeLR(optimizer, lambda epoch : decay_factor)
batchSize = 16
dataloader = data.DataLoader(dataset, batch_size=batchSize) 

for epoch in range(4): 
  print(optimizer.param_groups[0]['lr']) 
  for x in dataloader: 
      optimizer.step() 
  if epoch <= 0:
    scheduler1.step()
  else:
    scheduler2.step() 
```

```
0.0002
0.0004
0.0002
0.0001
```

**Comment by vincentqb** [comment]
You can also use LambdaLR as such:

```python
import torch
from torch import optim
from torch.utils import data

X = torch.arange(16) 
dataset = data.TensorDataset(X) 

optimizer = optim.Adam([torch.rand(10, 20),], lr=2e-4) 

warmup_factor = 2 
decay_factor = 0.5 

scheduler1 = optim.lr_scheduler.LambdaLR(optimizer, lambda epoch : warmup_factor**epoch)
scheduler2 = optim.lr_scheduler.LambdaLR(optimizer, lambda epoch : 1. if epoch <=0 else decay_factor ** (epoch-1))

batchSize = 16
dataloader = data.DataLoader(dataset, batch_size=batchSize) 

for epoch in range(4): 
  print(optimizer.param_groups[0]['lr']) 
  for x in dataloader: 
      optimizer.step() 
  if epoch <= 0:
    scheduler1.step()
  else:
    scheduler2.step() 
```

```
0.0002
0.0004
0.0002
0.0001
```

Notice that the warmup scheduler is defined outside of the epoch loop. I'll close this issue, but please feel free to re-open if the issue continues.

**Comment by vincentqb** [comment]
This can also be done with only one `MultiplicativeLR` (or similarly with one `LambdaLR`):

```python
import torch
from torch import optim
from torch.utils import data

X = torch.arange(16) 
dataset = data.TensorDataset(X) 

optimizer = optim.Adam([torch.rand(10, 20),], lr=2e-4) 

warmup_factor = 2 
decay_factor = 0.5 

scheduler = optim.lr_scheduler.MultiplicativeLR(optimizer, lambda epoch : warmup_factor if epoch <= 1 else decay_factor)

batchSize = 16
dataloader = data.DataLoader(dataset, batch_size=batchSize) 

for epoch in range(4): 
  print(optimizer.param_groups[0]['lr']) 
  for x in dataloader: 
      optimizer.step() 
  scheduler.step()
```

```
0.0002
0.0004
0.0002
0.0001
```
